### PR TITLE
spec: reorganize `basics` language specs

### PR DESCRIPTION
**Pull request by saem**
## Summary

Reorganize the `basics` language specification tests following the tone
of the `bootstrap` language specification tests. Tests are largely the
same but are now logically ordered based on a basic compiler
implementation.

## Details

Mostly files are reordered to follow a logical progression of
implementing a basic compiler or getting up to speed on language
basics.
Readme files were updated where appropriate to reflect the new style or
indicate the need for further updates.

Some minor stylistic changes were made to rely less on yet to be tested
language features, and the replacement of references to Nim to
Nimskull.

### CODE REVIEW

File: tests/lang/s00_bootstrap/t10_builtins_doAssert_bool_literals_expressions.nim [hunk 1]
```diff
@@ -16,8 +16,8 @@ block builtin_assert:
   doAssert true
 
 block tuatologies:
```
Comment (zerbina) [inline tests/lang/s00_bootstrap/t10_builtins_doAssert_bool_literals_expressions.nim]:
```suggestion
block tautologies:
```

File: tests/lang/s01_basics/s99_control_flow/readme.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+Language elements that can affect control flow of code execution.
+
+# Issues with This Section
+
+All these should not be tested here but later:
+- `try` and `defer`, advanced/syntatic sugar based constructs
+- `proc` are used but not defined under `basics`
+- `case` tightly relate to `enum`s and shouldn't be handled here
```
Comment (zerbina) [inline tests/lang/s01_basics/s99_control_flow/readme.md]:
```suggestion
- `case` tightly relates to `enum`s and shouldn't be handled here
```
Super minor, feel free to disregard.

**Review by zerbina — APPROVED** [review]
Looks good to me. I left two minor typo fix suggestions.

**Comment by saem** [comment]
thanks for the review, @zerbina

**Pull request merged**